GeoLib/AABB.h

```diff
@@ -70,9 +72,9 @@ class AABB
 	template <typename InputIterator>
 	AABB(InputIterator first, InputIterator last)
 	{
-		if (! (std::distance(first,last) > 0)) {
-			throw std::invalid_argument("AABB::AABB(InputIterator first, InputIterator last): first == last");
-		}
+		if (!(std::distance(first,last) > 0))
+			ERR("AABB::AABB(InputIterator first, InputIterator last): first > last");
+		assert(std::distance(first,last) > 0);
```

Comment:
Assuming that `std::distance(first, last) > 0`, does `assert()` something useful in release mode? On the other hand if the programm is not aborted at this place, it will either crash in the init function or will run forever in the while loop starting in line 80. Contrary, the previous solution was throwing an exception in release and debug mode, i.e. the same behaviour for debug and release mode.